Detectors/TPC/calibration/include/TPCCalibration/CalibPadGainTracks.h

```diff
@@ -0,0 +1,209 @@
+  /// e.g. mode=0 -> use the truncated mean from the track for normalizing the dedx
+  ///      mode=1 -> use the value from the BB-fit for normalizing the dedx. NOT implemented yet
+  void setMode(dEdxType iMode)
+  {
+    mMode = iMode;
+  }
+
+  /// initialize the histograms with default parameters
+  void initDefault()
+  {
+    mPadHistosDet = std::make_unique<o2::tpc::CalDet<o2::tpc::FastHisto<float>>>("Histo");
+  }
+
+  /// initialize the histograms with custom parameters
+  /// \param nBins number of bins used in the histograms
+  /// \param xmin minimum value in histogram
+  /// \param xmax maximum value in histogram
+  /// \param useUnderflow set usage of underflow bin
+  /// \param useOverflow set usage of overflow bin
+  void init(const unsigned int nBins, const float xmin, const float xmax, const bool useUnderflow, const bool useOverflow)
+  {
+    o2::tpc::FastHisto<float> hist(nBins, xmin, xmax, useUnderflow, useOverflow);
+    initDefault();
+    for (auto& calArray : mPadHistosDet->getData()) {
+      for (auto& tHist : calArray.getData()) {
+        tHist = hist;
+      }
+    }
+  }
+
+  /// dump the gain map to disk
+  void dumpGainMap();
+
+  /// get the truncated mean for each histogram and fill the extracted gainvalues in a CalPad object
+  void fillgainMap();
+
+  /// \return returns the gainmap object
+  CalPad getPadGainMap() const
+  {
+    return mGainMap;
+  }
+
+ private:
+  std::vector<o2::tpc::TrackTPC>* mTracks{nullptr};           ///< vector containing the tpc tracks which will be processed. Cant be const due to the propagate function
+  std::vector<TPCClRefElem>* mTPCTrackClIdxVecInput{nullptr}; ///< input vector with TPC tracks cluster indicies
+  const o2::tpc::ClusterNativeAccess* mClusterIndex{nullptr}; ///< needed to access clusternative with tpctracks
+  dEdxType mMode = DedxTrack;                                 ///< normalization type: type=DedxTrack use truncated mean, type=DedxBB use value from BB fit
+
+  inline static auto& mapper = Mapper::instance();     ///< initialize mapper object
+  static constexpr unsigned int NROWS = 152;           ///< number of padrows used TODO change to mapper
+  static constexpr unsigned int NROWSIROC = 63;        ///< number of padrows used TODO change to mapper
+  static constexpr unsigned int NROWSOROC = 89;        ///< number of padrows used TODO change to mapper
+  static constexpr unsigned int NSECTORS = 36;         ///< number of sectors TODO change to mapper
+  static constexpr unsigned int NPADSINSECTOR = 14560; ///< number of total pads in sector TODO change to mapper
+
+  std::unique_ptr<CalDet<o2::tpc::FastHisto<float>>> mPadHistosDet; ///< Calibration object containing for each pad a histogram with normalized charge
+  CalPad mGainMap{"GainMap"};                                       ///< Gain map object
+
+  /// calculate truncated mean for track
+  void processTrack(o2::tpc::TrackTPC& track, float momMin, float momMax);
```

Comment:
Since the function internally modifies `track` by calling `makeTrackTopologyCorrection`, it might be more appropriate to operate on a copy: `o2::tpc::TrackTPC track`. Otherwise, there might be problems if one does multiple computations on the same input data set.